vm/option.go

```diff
@@ -5,19 +5,46 @@ package vm
 
 import (
 	"github.com/ava-labs/hypersdk/builder"
+	"github.com/ava-labs/hypersdk/chain"
 	"github.com/ava-labs/hypersdk/gossiper"
+	"github.com/ava-labs/hypersdk/rpc"
+	"github.com/ava-labs/hypersdk/vm/event"
 )
 
-type Option func(*VM)
+type Option[T Controller] func(*VM[T])
 
-func WithManualBuilder() Option {
-	return func(vm *VM) {
+func WithManualBuilder[T Controller]() Option[T] {
+	return func(vm *VM[T]) {
 		vm.builder = builder.NewManual(vm)
 	}
 }
 
-func WithManualGossiper() Option {
-	return func(vm *VM) {
+func WithManualGossiper[T Controller]() Option[T] {
+	return func(vm *VM[T]) {
 		vm.gossiper = gossiper.NewManual(vm)
 	}
 }
+
+func WithBlockSubscriptions[T Controller](subscriptions ...event.SubscriptionFactory[*chain.StatelessBlock]) Option[T] {
```

Comment:
It matters in the sense that we call `Accept` in the order that the subscriptions are provided in. We could consider doing something like inserting these into an intermediary buffered channel instead of blocking the critical path of the hypersdk code.